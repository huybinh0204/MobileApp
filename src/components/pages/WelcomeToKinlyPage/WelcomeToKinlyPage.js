import { useFocusEffect, useNavigation } from '@react-navigation/native';
import React, { useCallback } from 'react';
import { MainButton } from '_components/atoms';
import { NAVIGATION } from '_constants';
import strings from '_localization';
import {
  Container,
  Content,
  Description,
  Illustration,
  Legal,
  Title,
} from './WelcomeToKinlyPage.styles';

const WelcomeToKinlyPage = () => {
  const { addListener, navigate } = useNavigation();

  const goToFundScreen = () => {
    navigate(NAVIGATION.auth.fundChoice);
  };

  useFocusEffect(
    useCallback(() => {
      return addListener('beforeRemove', (event) => {
        event.preventDefault(); // Disables go back
      });
    }, [addListener])
  );

  return (
    <Container edges={['bottom']} testID="WelcomeToKinlyPage">
      <Illustration source={require('_assets/illustrations/welcome-to-kinly-illustration.jpg')} />
      <Content>
        <Title>{strings.signUp.welcomeToKinly.title}</Title>
        <Description>{strings.signUp.welcomeToKinly.description}</Description>
        <Legal>{strings.signUp.welcomeToKinly.legal}</Legal>
        <MainButton accessibilityLabel="Continue" onPress={goToFundScreen}>
          {strings.signUp.continue}
        </MainButton>
      </Content>
    </Container>
  );
};

export default WelcomeToKinlyPage;
