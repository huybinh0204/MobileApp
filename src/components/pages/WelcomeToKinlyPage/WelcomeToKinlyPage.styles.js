import styled from '@emotion/native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { BaseText } from '_components/atoms';

export const Container = styled(SafeAreaView)(({ theme }) => ({
  flex: 1,
  backgroundColor: theme.colors.background,
}));

export const Illustration = styled.Image({
  width: '100%',
});

export const Content = styled.View(({ theme }) => ({
  flex: 1,
  alignItems: 'center',
  padding: theme.spacing.m,
}));

export const Title = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading4,
  color: theme.colors.beta900,
  marginBottom: theme.spacing.s,
  textAlign: 'center',
}));

export const Description = styled(BaseText)(({ theme }) => ({
  ...theme.typography.subContentRegular,
  color: theme.colors.beta500,
  marginBottom: 'auto',
  paddingHorizontal: theme.spacing.s,
  textAlign: 'center',
}));

export const Legal = styled(BaseText)(({ theme }) => ({
  ...theme.typography.legal,
  color: theme.colors.beta500,
  marginBottom: theme.spacing.s,
  textAlign: 'center',
}));
