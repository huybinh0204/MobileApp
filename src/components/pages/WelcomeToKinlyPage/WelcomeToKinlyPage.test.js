import { useNavigation } from '@react-navigation/native';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import React from 'react';
import { NAVIGATION } from '_constants';
import strings from '_localization/index';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import { navigationMock } from '_test/mocks/navigation';
import WelcomeToKinlyPage from './WelcomeToKinlyPage';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: jest.fn(),
}));

describe('WelcomeToKinlyPage', () => {
  const mockedNavigation = navigationMock(jest.fn);

  beforeEach(() => {
    useNavigation.mockReturnValue(mockedNavigation);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const { toJSON } = render(<ComponentWithProviders component={WelcomeToKinlyPage} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should navigate to the Fund screen', async () => {
    const { getByText } = render(<ComponentWithProviders component={WelcomeToKinlyPage} />);

    const continueButton = getByText(strings.signUp.continue);

    await waitFor(() => {
      fireEvent.press(continueButton);
    });

    expect(mockedNavigation.navigate).toBeCalledWith(NAVIGATION.auth.fundChoice);
  });
});
