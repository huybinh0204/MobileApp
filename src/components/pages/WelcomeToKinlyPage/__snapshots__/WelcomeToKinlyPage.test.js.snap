// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeToKinlyPage should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RNCSafeAreaView
                        edges={
                          Array [
                            "bottom",
                          ]
                        }
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#FAFAFA",
                              "flex": 1,
                            },
                            undefined,
                          ]
                        }
                        testID="WelcomeToKinlyPage"
                      >
                        <Image
                          source={
                            Object {
                              "testUri": "../../../src/assets/illustrations/welcome-to-kinly-illustration.jpg",
                            }
                          }
                          style={
                            Array [
                              Object {
                                "width": "100%",
                              },
                              undefined,
                            ]
                          }
                        />
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "flex": 1,
                                "padding": 58,
                              },
                              undefined,
                            ]
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#1F1D25",
                                  "fontFamily": "Poppins-SemiBold",
                                  "fontSize": 43.5,
                                  "lineHeight": 60,
                                  "marginBottom": 29,
                                  "textAlign": "center",
                                },
                                undefined,
                              ]
                            }
                          >
                            Welcome to Kinly
                          </Text>
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#5F6A77",
                                  "fontFamily": "Poppins",
                                  "fontSize": 25.5,
                                  "lineHeight": 35.5,
                                  "marginBottom": "auto",
                                  "paddingHorizontal": 29,
                                  "textAlign": "center",
                                },
                                undefined,
                              ]
                            }
                          >
                            Your Kinly debit card is on its way! You should expect to see it in your mail within 8-10 days, but sometimes even faster.
                          </Text>
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#5F6A77",
                                  "fontFamily": "Poppins",
                                  "fontSize": 20,
                                  "lineHeight": 28,
                                  "marginBottom": 29,
                                  "textAlign": "center",
                                },
                                undefined,
                              ]
                            }
                          >
                            The Kinly VisaÂ® Debit Card is issued 
by the Bancorp Bank, Member FDIC.
                          </Text>
                          <View
                            accessibilityLabel="Continue"
                            accessibilityState={
                              Object {
                                "disabled": false,
                              }
                            }
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignSelf": "stretch",
                                "backgroundColor": "#3D3397",
                                "borderRadius": 10,
                                "opacity": 1,
                                "paddingHorizontal": 45.5,
                                "paddingVertical": 31,
                              }
                            }
                          >
                            <Text
                              allowFontScaling={false}
                              disabled={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#FFFFFF",
                                    "fontFamily": "Poppins-SemiBold",
                                    "fontSize": 25.5,
                                    "lineHeight": 35.5,
                                    "textAlign": "center",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Continue
                            </Text>
                          </View>
                        </View>
                      </RNCSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
